dxr/static/js/code-highlighter.js

```diff
@@ -115,21 +115,25 @@ $(function () {
         }
     }
 
-    //update places where hash location is used: window, permalink
+    //update places where hash location is used: window, sync-links
     function updateHash(hash) {
-        if (permalink.length > 0)
-            updatePermalink(hash);
+        syncLinks.each(function() {
+            updateLink($(this), hash);
+        });
         history.replaceState(null, '', hash);
     }
 
-    //update the permalink href based on the windowHash.
-    function updatePermalink(windowHash) {
-        var permalink_href = permalink.attr('href'),
-            hash_loc = permalink_href.indexOf('#');
-        // If the link already has #, then cut that it off.
+    //update the link href based on the windowHash.
+    function updateLink(anchor, windowHash) {
+        var href = anchor.attr('href'),
+            hash_loc = href.indexOf('#');
+        // If the link already has #, then cut it off.
         if (hash_loc >= 0)
-            permalink_href = permalink_href.substring(0, hash_loc);
-        permalink.attr('href', permalink_href + windowHash);
+            href = href.substring(0, hash_loc);
+        // hg web uses #l and GitHub uses #L but supports #l, so we use #l here.
```

Comment:
That's cool. If, instead of just tacking the anchor on at the end, we did some sort of token substitution, we could support sending entire highlight *ranges* to GitHub. For example, we might use something like "{start_highlight}" and "{end_highlight}", which would let us express URLs to systems that support just a single line (like hgweb) or multiple (like GH) without a lot of special casing. For example, GH would be "/...#L{start_highlight}-L{end_highlight}".